src/scaffoldmaker/meshtypes/meshtype_3d_lung2.py

```diff
@@ -0,0 +1,2139 @@
+from scaffoldmaker.utils.meshrefinement import MeshRefinement
+from scaffoldmaker.utils.vector import magnitude, setMagnitude, crossproduct3, normalise
+from opencmiss.utils.zinc.field import Field, findOrCreateFieldCoordinates, findOrCreateFieldGroup, \
+    findOrCreateFieldNodeGroup, findOrCreateFieldStoredMeshLocation, findOrCreateFieldStoredString
+from opencmiss.utils.zinc.finiteelement import get_element_node_identifiers
+from opencmiss.zinc.element import Element
+from opencmiss.zinc.node import Node
+
+class MeshType_3d_lung2(Scaffold_base):
+    '''
+    Generic 3D lung scaffold.
+    '''
+
+    """
+        Generates an ellipsoid with a tear-shaped base for the lung mathematically,
+         with x, y, z length and the position, angle of the oblique fissure.
+         Regions and markers of the lung are annotated.
+    """
+
+    @staticmethod
+    def getName():
+        return '3D Lung 2'
+
+    @staticmethod
+    def getParameterSetNames():
+        return [
+            'Default',
+            'Human 1',
+            'Mouse 1',
+            'Rat 1',
+            'Pig 1',
+            'Material'
+        ]
+
+    @classmethod
+    def getDefaultOptions(cls, parameterSetName='Default'):
+        if parameterSetName == 'Default':
+            parameterSetName = 'Human 1'
+        options = {
+            'Base parameter set': parameterSetName,
+            'Number of left lung lobes': 2,
+            'Left-right lung spacing': 1.0,
+            'Left-right apex closing space': 0.0,
+            'Left-right apex along y-axis': 0.0,
+            'Left lung width': 0.25,
+            'Left lung depth': 0.5,
+            'Left lung height': 1.0,
+            'Left lung ventral edge sharpness factor': 0.0,
+            'Left lung medial curvature factor': 0.0,
+            'Left lung medial protrusion factor': 0.0,
+            'Right lung width': 0.25,
+            'Right lung depth': 0.5,
+            'Right lung height': 1.0,
+            'Right lung ventral edge sharpness factor': 0.0,
+            'Right lung medial curvature factor': 0.0,
+            'Right lung medial protrusion factor': 0.0,
+            'Open fissures': False,
+            'Accessory lobe': True,
+            'Accessory lobe medial curvature': 0.0,
+            'Accessory lobe length': 0.5,
+            'Accessory lobe dorsal centre [x,y]': [0.0, 0.0],
+            'Accessory lobe dorsal height': 0.5,
+            'Accessory lobe dorsal width': 0.5,
+            'Accessory lobe ventral height': 0.5,
+            'Accessory lobe ventral width': 0.5,
+            'Diaphragm centre x': 0.0,
+            'Diaphragm centre y': 0.0,
+            'Diaphragm curvature x': 0.0,
+            'Diaphragm curvature y': 0.0,
+            'Rotation in degree about [z, y, x]: Left lung': [0.0, 0.0, 0.0],
+            'Rotation in degree about [z, y, x]: Right lung': [0.0, 0.0, 0.0],
+            'Rotation in degree about [z, y, x]: Accessory lobe': [0.0, 0.0, 0.0],
```

Comment:
"Left lung rotation in degrees about z, y, z" etc. following wording for the general rotation parameter, and also put options in with the options for Left, Right and Accessory lobe above.
See comments later about these being applied in the opposite sense to that expected (see what the general rotation parameter does with a positive rotation, it's a positive right hand rotation about the respective axis).
It seems these rotations are about the origin (see comments with the function later).
Will it ever make sense to rotate about y or x? Rotations on those axes affects the curve of the diaphragm...

Reply:
I removed rotations about y and x-axis as you suggested and make it apply to a right-hand rule.
I also fixed the rotation at the centre of elements.